pkg/server/helpers.go

```diff
@@ -216,3 +223,129 @@ func (c *criContainerdService) getSandbox(id string) (*metadata.SandboxMetadata,
 func criContainerStateToString(state runtime.ContainerState) string {
 	return runtime.ContainerState_name[int32(state)]
 }
+
+// normalizeImageRef normalizes the image reference following the docker convention. This is added
+// mainly for backward compatibility.
+// The reference returned can only be either tagged or digested. For reference contains both tag
```

Comment:
@Random-Liu I see. And hence the problem with that model. ;)

I suppose you can split them into a tagged and digested reference (as I think is being done here).